src/main/java/net/geant/nmaas/portal/service/impl/BulkApplicationServiceImpl.java

```diff
@@ -246,9 +246,12 @@ public ApplicationEvent handleDeploymentStatusUpdate(AppAutoDeploymentStatusUpda
                     bulkDeploymentEntryRepository.save(bulkDeploymentEntry);
                     return new AppAutoDeploymentReviewEvent(this);
                 default:
-                    Thread.sleep(event.getWaitIntervalBeforeNextCheckInMillis() > 0 ?
-                            event.getWaitIntervalBeforeNextCheckInMillis() : WAIT_INTERVAL_IN_SECONDS * 1000);
-                    return event;
+                    int seconds = event.getWaitIntervalBeforeNextCheckInMillis() > 0 ?
+                            event.getWaitIntervalBeforeNextCheckInMillis() : WAIT_INTERVAL_IN_SECONDS * 1000;
+                    Thread.sleep(seconds);
+                    event.setEventTimeOutMinutes(event.getEventTimeOutMinutes() - (seconds / 60));
```

Comment:
Probably it would be better to define eventTimeOut in seconds not in minutes. Also please add a log on debug level to display the time remaining till timeout.